pkg/catalog/inbound_traffic_policies.go

```diff
@@ -68,6 +50,26 @@ func (mc *MeshCatalog) GetInboundMeshTrafficPolicy(upstreamIdentity identity.Ser
 
 		upstreamTrafficSetting := mc.GetUpstreamTrafficSettingByService(&upstreamSvc)
 		clusterConfigs = append(clusterConfigs, getRateLimitServiceClusters(upstreamTrafficSetting, rlsClusterSet)...)
+	}
+
+	return clusterConfigs
+}
+
+// GetInboundMeshTrafficMatches returns the traffic matches for the inbound mesh traffic policy for the given upstream services
+func (mc *MeshCatalog) GetInboundMeshTrafficMatches(upstreamServices []service.MeshService) []*trafficpolicy.TrafficMatch {
+	allUpstreamServices := mc.getUpstreamServicesIncludeApex(upstreamServices)
```

Comment:
Looking at this a bit more it looks like this:

1. has 2 vars, `upstreamServices`, and `allUpstreamServices`, which adds apex services.
2. upstreamSvcSet is comprised only of `upstreamServices`.
3. iterates the list of `allUpstreamServices`, but skips if it's not in upstreamServicesSet.

That means, we never need to call `getUpstreamServicesIncludeApex`, and we can just iterate `upstreamServices`.

It also means we never need the `upstreamSvcSet`